cleanlab/multiannotator.py

```diff
@@ -0,0 +1,859 @@
+
+        def get_single_annotator_quality(s, consensus_label_subset, quality_of_consensus_subset):
+            annotator_mask = pd.notna(s)
+            try:
+                annotator_quality = np.average(
+                    s[annotator_mask] == consensus_label_subset[annotator_mask],
+                    weights=quality_of_consensus_subset[annotator_mask],
+                )
+            except:  # if annotator does not overlap with other annotators, return NaN as annotator quality score
+                annotator_quality = np.NaN
+
+            return annotator_quality
+
+        annotator_quality = labels_multiannotator_subset.apply(
+            lambda s: get_single_annotator_quality(
+                s, consensus_label_subset, quality_of_consensus_subset
+            ),
+            axis=0,
+        )
+
+    elif quality_method == "agreement":
+        mask = num_annotations != 1
+        labels_multiannotator_subset = labels_multiannotator[mask]
+        consensus_label_subset = consensus_label[mask]
+
+        annotator_quality = labels_multiannotator_subset.apply(
+            lambda s: np.mean(s[pd.notna(s)] == consensus_label_subset[pd.notna(s)]),
+            axis=0,
+        )
+
+    else:
+        raise ValueError(
+            f"""
+            {quality_method} is not a valid annotator quality method!
+            Please choose a valid quality_method: {valid_methods}
+            """
+        )
+
+    return annotator_quality
+
+
+def get_annotator_stats(
+    labels_multiannotator: pd.DataFrame,
+    pred_probs: np.ndarray,
+    consensus_label: np.ndarray,
+    num_annotations: np.ndarray,
+    annotator_agreement: np.ndarray,
+    quality_of_consensus: np.ndarray,
+    quality_method: str = "auto",
+) -> pd.DataFrame:
+    """Returns overall statistics about each annotator.
+
+    Parameters
+    ----------
+    labels_multiannotator : pd.DataFrame
+        2D pandas DataFrame of multiple given labels for each example with shape (N, M),
+        where N is the number of examples and M is the number of annotators.
+        For more details, labels in the same format expected by the :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`.
+    pred_probs : np.ndarray
+        An array of shape ``(N, K)`` of model-predicted probabilities, ``P(label=k|x)``.
+        For details, predicted probabilities in the same format expected by the :py:func:`get_label_quality_multiannotator <cleanlab.multiannotator.get_label_quality_multiannotator>`.
+    consensus_label : np.ndarray
+        An array of shape ``(N,)`` with the consensus labels aggregated from all annotators.
+    num_annotations : np.ndarray
+        An array of shape ``(N,)`` with the number of annotators that have labeled each example.
+    annotator_agreement : np.ndarray
+        An array of shape ``(N,)`` with the fraction of annotators that agree with each consensus label.
+    quality_method : str
```

Comment:
In other parts of codebase they specify if there is a default method in docstring 
i.e. quality_method : str, default = "auto"